compiler/msgs.nim

```diff
@@ -480,9 +480,17 @@ proc sourceLine*(conf: ConfigRef; i: TLineInfo): string =
 proc getSurroundingSrc(conf: ConfigRef; info: TLineInfo): string =
```

Comment:
how about add an inplace overload:
```nim
proc getSurroundingSrc(result: var string, conf: ConfigRef; info: TLineInfo) = ...
proc getSurroundingSrc(conf: ConfigRef; info: TLineInfo): string = getSurroundingSrc(result, conf, info)
```
and then just used append ops

Reply:
I think I get what you mean though don't see much benefit, the bigger concern to me presently is applying some heursitics to selecting only the pertinent information in the source hint. Though that gets rather complex rather quickly and may hide information.
Presently some messages look like this 
![image](https://user-images.githubusercontent.com/11792483/124845818-b9a86580-df54-11eb-99ae-dfbd4dafb710.png)
though helpful for this kind
![image](https://user-images.githubusercontent.com/11792483/124845833-c75deb00-df54-11eb-9f73-9dd4f98daa23.png)

Reply:
> I think I get what you mean though don't see much benefit

just ignore it for now then (it's for performance reason).

> the bigger concern to me presently is applying some heursitics to selecting only the pertinent information in the source hint.

the problem is we're not tracking an interval in the source; TLineInfo only captures a point instead of interval (unlike tools like clang etc which can thus generate nicer diagnostics). I think we'll have to address this at some point but for this PR, how about simply colorize the next token starting from TLineInfo; i think it's a simple enough heuristic.

using lexer should tell you where the token ends, or maybe using a heuristic can work well (eg stop at whitespace or non-identifier character), although it'd require care for tokens of type stringlit or number literals, which is why re-using lexer would be the simplest/most robust. Let me know if that makes sense.

Reply:
Yea it all makes sense, I mainly forgot about the fact the source can be printed out more than once which is why the confusion due to the in place version.